src/yamux/yamux.lua

```diff
@@ -176,17 +177,41 @@ proto_yamux.prefs.heuristic = Pref.bool("Heuristic enabled", default_settings.he
     "Whether heuristic dissection is enabled.")
 proto_yamux.prefs.debug = Pref.enum("Debug level", default_settings.debug_level, "The debug level verbosity",
     util.debug.pref_enum)
+proto_yamux.prefs.ports = Pref.range("Port(s)", default_settings.ports,
+    "Set the TCP port(s) for Yamux, default is 443. That ports are used only with disabled heuristics analysis", 65535)
+
+-- Got from zabbixagent
+local function enableDissector()
+    if not default_settings.heuristic_enabled then
+        DissectorTable.get("tcp.port"):add(default_settings.ports, proto_yamux)
+        -- supports also TLS decryption if the session keys are configured in Wireshark
+        DissectorTable.get("tls.port"):add(default_settings.ports, proto_yamux)
+    end
+end
+-- call it now, because we're enabled by default
+enableDissector()
+
+local function disableDissector()
+    if not default_settings.heuristic_enabled then
+        DissectorTable.get("tcp.port"):remove(default_settings.ports, proto_yamux)
+        DissectorTable.get("tls.port"):remove(default_settings.ports, proto_yamux)
+    end
+end
 
 --- Handle changes to Wireshark preferences for this protocol
 -- @function proto_yamux.prefs_changed
 function proto_yamux:prefs_changed()
     dprint2("prefs_changed called")
+    disableDissector()
```

Comment:
```suggestion
    -- Disable the dissector if heuristic detection is disabled.
    -- The dissector will be re-enabled if the heuristic detection is still
    -- disabled after processing the preference changes.
    disableDissector()
```

The ` if not default_settings.heuristic_enabled then` conditional in both functions was a little confusing when I first looked at these functions. I think adding a comment like this will be helpful in clarifying what happens during  the `prefs_changed()` callback.